### PR TITLE
Save button refactor

### PR DESCRIPTION
**Pull request by kapoliub**
Issue:
https://github.com/ita-social-projects/horondi_client_fe/issues/469

Also fixed bug when admin clicked few times at safe button (all fields are valid) and recieved popup about error (admin saw new material/pattern only after page reloading). Now button is disabled after save button clicking.

### CODE REVIEW

File: src/components/buttons/save-button/save-button.js [hunk 1]
```diff
@@ -1,29 +1,72 @@
...
+}) => {
+  const { IMG_URL } = config;
+  const error = !!Object.keys(errors).length;
+  const disable = Object.values(values).every((el) => {
+    console.log(el);
+    if (typeof el === 'boolean') {
+      return true;
+    }
+    if (el) {
+      if (el === IMG_URL) {
```
Comment (ivan-bonk) [inline src/components/buttons/save-button/save-button.js:24]: We don't need this IMG_URL in categories form

File: src/components/buttons/save-button/save-button.js [hunk 2]
```diff
@@ -1,29 +1,72 @@
...
+  onClickHandler,
+  color,
+  errors,
+  values,
+  ...props
+}) => {
+  const { IMG_URL } = config;
+  const error = !!Object.keys(errors).length;
+  const disable = Object.values(values).every((el) => {
+    console.log(el);
```
[the remark this answers is not in the document]
Comment (kapoliub) [reply src/components/buttons/save-button/save-button.js]: fixed

File: src/components/buttons/save-button/save-button.js (hunk 1 above)
Comment (kapoliub) [reply src/components/buttons/save-button/save-button.js]: fixed

File: src/components/category-form/category-form.js [hunk 3]
```diff
@@ -82,7 +81,7 @@ const CategoryForm = ({ category, id, edit }) => {
   } = useFormik({
     validationSchema: categoryValidationSchema,
     initialValues: {
-      categoryImage: IMG_URL + category.images.thumbnail || '',
+      categoryImage: category.images.thumbnail || '',
```
Comment (moran711) [inline src/components/category-form/category-form.js:84]: Why you remove possibility to see photo when we edit category?

**Review by moran711 — APPROVED** [review]

File: src/components/category-form/category-form.js (hunk 3 above)
Comment (kapoliub) [reply src/components/category-form/category-form.js]: fixed

**Review by ivan-bonk — APPROVED** [review]

**Review by NikitaDenysenko — APPROVED** [review]

**Pull request closed without merging**